obstacle_stop_planner_refine/include/obstacle_stop_planner/obstacle_stop_planner.hpp

```diff
@@ -42,34 +39,56 @@
 
 namespace obstacle_stop_planner
 {
+using Trajectory = autoware_planning_msgs::msg::Trajectory;
+using Point3d = autoware_utils::Point3d;
+using LinearRing2d = autoware_utils::LinearRing2d;
+struct Input
+{
+  geometry_msgs::msg::Pose current_pose;
+  std::vector<Point3d> obstacle_pointcloud;
+  rclcpp::Time pointcloud_header_time;
+  geometry_msgs::msg::TwistStamped current_velocity;
+  autoware_perception_msgs::msg::DynamicObjectArray object_array;
+  Trajectory input_trajectory;
+};
+
+struct Output
+{
+  Trajectory output_trajectory;
+  visualization_msgs::msg::MarkerArray debug_viz_msg;
+  autoware_planning_msgs::msg::StopReasonArray stop_reason;
+  autoware_debug_msgs::msg::Float32MultiArrayStamped acc_debug_msg;
+};
+
+struct Collision
+{
+  size_t trajectory_index;
```

Comment:
I think `segment_index` is better.
Then we can know `trajectory.,at(segment_index)` is always before the `obstacle_point`.

black: point index
red: segment index
![image](https://user-images.githubusercontent.com/31987104/114882914-659f5080-9e3f-11eb-8a99-40206cae592f.png)

Reply:
That's a good idea!